### PR TITLE
Introduce tests, use DateUtil class, fixes bug #48

### LINKED ISSUE #48: [Core Weekly ] Bug in rendering dates, doesn't handle 2 months
Phrase
```
This edition of the Core Weekly report highlights changes in PrestaShop's core codebase from Monday 7th to Sunday 13th of September 2020.
```

If 1st day and 2nd day are not same month (example 31/08 and 06/09) it will print only once the month.

It will print:
```
This edition of the Core Weekly report highlights changes in PrestaShop’s core codebase from Monday 31th to Sunday 6th of September 2020.
```

It should be
```
This edition of the Core Weekly report highlights changes in PrestaShop’s core codebase from Monday 31th August to Sunday 6th of September 2020.
```

I think same will happen with years 😄

### PR DESCRIPTION
**Pull request by matks**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Description?  | In this PR I extract some date functions and unit test them. I also fix a bug.
| Type?         | improvement
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | Fixes https://github.com/PrestaShop/core-weekly-generator/issues/48
| How to test?  | CI should pass

### CODE REVIEW

**Comment by matks** [comment]
CI succeed 🎉 https://github.com/matks/core-weekly-generator/runs/1162673361

**Comment by matks** [comment]
CI (updated) successfull https://github.com/matks/core-weekly-generator/runs/1170218789

File: .github/workflows/nose.yml [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+    steps:
+    - uses: actions/checkout@v2
+    - name: Set up Python ${{ matrix.python-version }}
+      uses: actions/setup-python@v2
+      with:
+        python-version: ${{ matrix.python-version }}
+    - name: Install dependencies
+      run: |
+        python -m pip install --upgrade pip
+        pip install flake8 pytest nose
```
Comment (PierreRambaud) [inline .github/workflows/nose.yml:22]:
```suggestion
```
Puts them in requirements-dev.txt or requirements.txt

File: .github/workflows/nose.yml [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+        if [ -f requirements.txt ]; then pip install -r requirements.txt; fi
```
Comment (PierreRambaud) [inline .github/workflows/nose.yml]:
```suggestion
        python -m pip install --upgrade pip
        setup.py install
```
Should be enough. Dependecies are already declared in the setup.py files

File: README.md [hunk 3]
```diff
@@ -71,3 +71,19 @@ You need to do a bit of polishing (fill in the `[XXXX]` sections, reorder catego
 ## Troubleshooting
 
 Be aware that data fetched from GitHub is stored into a local sqlite cache file! This might create issues when submitting code changes to the tool logic.
+
+## Running tests
+
+Install dependencies:
+
+```bash
+$ ./setup.py test
```
Comment (PierreRambaud) [inline README.md]:
```suggestion
$ ./setup.py install
```

**Review by PierreRambaud — APPROVED** [review]

**Comment by matks** [comment]
Very nice work @PierreRambaud thanks !

**Pull request merged**